xarray/core/iterators.py

```diff
@@ -1,6 +1,9 @@
+from __future__ import annotations
+
 from abc import abstractmethod
 from collections import abc
-from typing import Callable, Iterator, List, Optional
+from collections.abc import Iterator
```

Comment:
This change looked a bit unexpected, but `typing.Iterator` is [a deprecated alias for collections.abc.Iterator](https://docs.python.org/3/library/typing.html#typing.Iterator)